### DISCUSSION TITLE
What makes this number pop up out of nowhere?

### DISCUSSION THREAD

**Discussion by Tunaik** [community/community#21406] [Other Feature Feedback, Questions, & Ideas]
<p>Hi so I’m currently learning C and i’ve accidently found out something while playing with variables.<br>
After declaring and initializing the value “age” I’ve tried using it on a print command, making the format<br>
specified as %d , but I’ve forgotten to include the code “,age” at the end of the print command and this<br>
random number popped up (Value for the age was 21), I’m kinda newbie at coding and got curious about what this number might be<br>
Any ideas?<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/d/6d2eca184a3544839b9d143a1865bdda5ff70817.png" data-download-href="/uploads/short-url/fzShZrkrcg1LzQBAoL5mWwbO1mv.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/106908525/181064438-93fd9d6d-a6ef-4335-9316-43f401bd51e3.png" alt="image" data-base62-sha1="fzShZrkrcg1LzQBAoL5mWwbO1mv" width="490" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/d/6d2eca184a3544839b9d143a1865bdda5ff70817_2_490x500.png, https://aws1.discourse-cdn.com/github/original/3X/6/d/6d2eca184a3544839b9d143a1865bdda5ff70817.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/6/d/6d2eca184a3544839b9d143a1865bdda5ff70817.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/d/6d2eca184a3544839b9d143a1865bdda5ff70817_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">648×660 41.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mpboom** [comment]
<p>If you call <code>printf</code> with too little arguments, it’s <a href="https://stackoverflow.com/questions/28194227/printf-with-no-arguments-in-c-compiles-fine-how" rel="noopener nofollow ugc">behavior is undefined</a>.</p>

**Comment by Tunaik** [comment]
<p>I’m aware of that, but shouldnt it give null value instead of the number “199…” I don’t understand why it specifically gives that value</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That depends on your compiler and C library. “Undefined behavior” means they can do anything. Give some number, crash, draw a bunny on the screen… anything. In practice it’s most likely an artifact of how <code>printf()</code> is implemented. :sweat_smile:</p>
<p>See also: <a href="https://en.wikipedia.org/wiki/Undefined_behavior" class="inline-onebox">Undefined behavior - Wikipedia</a></p>
